### PR TITLE
CTCTRADERS-895 Navigation for Trader route 

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/navigation/Navigator.scala [hunk 1]
```diff
@@ -72,5 +126,65 @@ class Navigator @Inject()() {
...
+  def principalEoriKnown(mode: Mode, ua: UserAnswers): Call = {
+    (ua.get(IsPrincipalEoriKnownPage), mode) match {
+      case (Some(true), NormalMode) => routes.WhatIsPrincipalEoriController.onPageLoad(ua.id, NormalMode)
+      case (Some(true), CheckMode) => routes.WhatIsPrincipalEoriController.onPageLoad(ua.id, CheckMode)
+    }
+  }
+
+  def principalEoriNotKnown(mode: Mode, ua: UserAnswers): Call = {
+    (ua.get(IsPrincipalEoriKnownPage), mode) match {
+      case (Some(false), NormalMode) => routes.PrincipalNameController.onPageLoad(ua.id, NormalMode)
```
Comment (ashwinirudrappa) [inline app/navigation/Navigator.scala:140]: What should happen when for Some(true) and None case

File: app/navigation/Navigator.scala [hunk 2]
```diff
@@ -72,5 +126,65 @@ class Navigator @Inject()() {
       case _ => movementDetailsRoutes.MovementDetailsCheckYourAnswersController.onPageLoad(ua.id)
     }
   }
+
+
+  def principalEoriKnown(mode: Mode, ua: UserAnswers): Call = {
+    (ua.get(IsPrincipalEoriKnownPage), mode) match {
+      case (Some(true), NormalMode) => routes.WhatIsPrincipalEoriController.onPageLoad(ua.id, NormalMode)
```
Comment (ashwinirudrappa) [inline app/navigation/Navigator.scala:133]: What should happen to Some(true) and None case

File: app/utils/SectionsHelper.scala [hunk 3]
```diff
@@ -68,11 +69,12 @@ class SectionsHelper(userAnswers: UserAnswers) {
 
   private def tradersDetailsSection: SectionDetails = {
     val startPage: String = routes.IsPrincipalEoriKnownController.onPageLoad(userAnswers.id, NormalMode).url
-    val cyaPageAndStatus: (String, Status) = (movementDetailsRoutes.MovementDetailsCheckYourAnswersController.onPageLoad(userAnswers.id).url, Completed) //TODO specific check your answers
+    val cyaPageAndStatus: (String, Status) = (routes.TraderDetailsCheckYourAnswersController.onPageLoad(userAnswers.id).url, Completed) //TODO specific check your answers
```
Comment (ashwinirudrappa) [inline app/utils/SectionsHelper.scala:72]: @kenneth-rayner The TODO is already addressed can we remove this TODO

File: conf/messages.en [hunk 4]
```diff
@@ -253,20 +253,16 @@ whatIsPrincipalEori.error.invalidFormat = Enter the principal’s EORI number in
 
 principalAddress.title = What is the principal’s address?
 principalAddress.heading = What is {0}’s address?
-principalAddress.country = Country
-principalAddress.checkYourAnswersLabel =  Principal’s address
-
 principalAddress.numberAndStreet = Number and street
 principalAddress.town = Town
 principalAddress.postcode = Postcode
+principalAddress.checkYourAnswersLabel =  Principal’s address
```
Comment (ashwinirudrappa) [inline conf/messages.en:259]: Don't we need the below messages it has been removed, please take a look 

principalAddress.error.numberAndStreet.invalid = The  number and street of {0}'s address must only include letters a to z, numbers 0 to 9, ampersands (&), apostrophes, asterisks, forward slashes, full stops, hyphens, question marks, and greater than (>) and less than (<) signs		
 principalAddress.error.town.invalid = The town of {0}'s address must only include letters a to z, numbers 0 to 9, ampersands (&), apostrophes, asterisks, forward slashes, full stops, hyphens, question marks, and greater than (>) and less than (<) signs

File: conf/messages.en [hunk 5]
```diff
@@ -344,6 +339,9 @@ countryOfDispatch.heading = What is the country of dispatch?
 countryOfDispatch.checkYourAnswersLabel = Country of dispatch
 countryOfDispatch.error.required = Select the country of dispatch
 
+traderDetailsCheckYourAnswers.title = Check your answers
```
Comment (ashwinirudrappa) [inline conf/messages.en:342]: Can we use the existing label for Check your answers

**Review by ashwinirudrappa — COMMENTED** [review]
Please address the relevant comments

File: app/navigation/Navigator.scala [hunk 6]
```diff
@@ -38,15 +38,40 @@ class Navigator @Inject()() {
...
     case _ => _ => routes.IndexController.onPageLoad()
   }
 
   private val checkRouteMap: Page => UserAnswers => Call = {
     case DeclarationForSomeoneElsePage => ua => isDeclarationForSomeoneElse(ua, CheckMode)
     case page if isMovementDetailsSectionPage(page) => ua => movementDetailsRoutes.MovementDetailsCheckYourAnswersController.onPageLoad(ua.id)
+    case IsPrincipalEoriKnownPage => ua => isPrincipalEoriKnownRoute(ua, CheckMode)
+    case PrincipalNamePage => ua => principalNamePageRoute(ua, CheckMode)
+    case ConsignorNamePage => ua => consignorNamePageRoute(ua, CheckMode)
+    case ConsigneeNamePage => ua => consigneeNamePageRoute(ua, CheckMode)
```
[the remark this answers is not in the document]
Comment (kenneth-rayner) [reply app/navigation/Navigator.scala:64]: One says consignee one says consignor

**Review by ashwinirudrappa — APPROVED** [review]

**Pull request merged**